src/server/saved_objects/client/saved_objects_client.js

```diff
@@ -173,24 +215,33 @@ export class SavedObjectsClient {
       return { saved_objects: [] };
     }
 
-    const docs = objects.reduce((acc, { type, id }) => {
-      return [...acc, {}, createIdQuery({ type, id })];
-    }, []);
-
-    const response = await this._withKibanaIndex('msearch', { body: docs });
-    const responses = get(response, 'responses', []);
+    const response = await this._withKibanaIndex('mget', {
+      body: {
+        docs: objects.map(object => ({
+          _id: this._generateEsId(object.type, object.id),
+          _type: V6_TYPE,
+        }))
+      }
+    });
 
     return {
-      saved_objects: responses.map((r, i) => {
-        const [hit] = get(r, 'hits.hits', []);
+      saved_objects: response.docs.map((doc, i) => {
+        const { id, type } = objects[i];
 
-        if (!hit) {
-          return Object.assign({}, objects[i], {
+        if (doc.found === false) {
+          return {
+            id,
+            type,
             error: { statusCode: 404, message: 'Not found' }
```

Comment:
We don't reflect the status of individual requests in the HTTP status since this is basically an mget, but if you're thinking something like `Boom.notFound().output.payload` I'm indifferent. Seems sort of like overkill though.